### ISSUE TITLE
on_startup / on_shutdown

### ISSUE THREAD

**Issue by SandaruKasa** [pyrogram/pyrogram#1264]
### Checklist

- [X] I believe the idea is awesome and would benefit the framework
- [X] I have searched in the issue tracker for similar requests, including closed ones

### Description

It would be nice to have an ability to run some custom code when the client starts/stops.

[The recommended way](https://docs.pyrogram.org/start/invoking#context-manager) of using `client.run()` currently does not allow to run any code after the client is started or before it is stopped.

I guess it still possible to
```python
async def main():
    client = ...
    async with client:
        await on_startup(client)
        await pyrogram.methods.utilities.idle.idle()
        await on_shutdown(client)

asyncio.run(main())
```
but this has a drawback of you having to initialize `client` inside of the `main`, as mentioned [here](https://docs.pyrogram.org/start/invoking#using-asyncio-run).

My guess is that it is possible to escape the need to do so by using the same code as `client.run()` itself uses:
```python
client = ...
def main():
    loop = asyncio.get_event_loop()
    run = loop.run_until_complete
    run(client.start())
    run(pyrogram.methods.utilities.idle.idle())
    run(on_startup(client))
    run(on_shutdown(client))

main()
```
but this is getting too hairy.

What I propose is something similar to aiogram's [`on_startup` & `on_shutdown`](https://github.com/aiogram/aiogram/blob/v2.25.1/aiogram/utils/executor.py#L25-L27).

So that now you would be able to just
```python
client = ...

client.run(
    on_startup=on_startup,
    on_shutdown=on_shutdown,
)
```

**Comment by SandaruKasa** [comment]
Well, I guess it is possible to
```python
client = ...
async def main(client):
    await on_startup(client)
    await pyrogram.methods.utilities.idle.idle()
    await on_shutdown(client)

client.run(main())
```
but this still doesn't look that pleasant.

Also, considering that `compose` is a thing and pyrogram allows running multiple clients in parallel, I guess it would make more sense to implement `on_startup` and `on_shutdown` as client's handlers instead of just some arguments you pass to the `run` function.
